lib/afc/streams.js

```diff
@@ -20,21 +20,23 @@ class AfcReadableFileStream extends stream.Readable {
     }
 
     if (_.isEmpty(data)) {
-      await this._destroy();
+      this.destroy();
       return;
     }
 
     this.push(data);
   }
 
-  async _destroy () {
+  async _destroy (err, cb) {
     if (this._destroyed) {
       return;
     }
     this._destroyed = true;
-    await this._afcService.closeFileHandle(this._fileHandle);
     this.push(null);
-    this.emit('close');
+    try {
+      await this._afcService.closeFileHandle(this._fileHandle);
+    } catch (ign) {}
+    cb(err);
```

Comment:
should we check if `cb` is a function here?

Reply:
The documentation says it will be always a function